tests/jeos/firstrun.pm

```diff
@@ -67,7 +67,10 @@ sub verify_mounts {
 
 sub run {
     my ($self) = @_;
-
+    # kiwi-templates-JeOS images (sle, opensuse x86_64 only) are build w/o translations
+    # jeos-firstboot >= 0.0+git20200827.e920a15 locale warning dialog has been removed
+    my $jeos_first_tag = (is_sle('<15-sp3') || (is_opensuse && is_x86_64)) ? 'jeos-lang-warning' :
```

Comment:
The variable is only used once, so just calling `assert_screen` directly is more direct. `if`/`elseif`/`else` is also easier to read

Reply:
That would require to use an array of tags, which would prolong the execution time (more needles to evaluate). Also, this way get little more control over if the updated jeos-firstboot w/o the informative pop up reached expected distros (as of now only sle15sp3 got the newest jeos-firstboot).
And my main motivation was to isolate the part of the code and needle tag which will become obsolete and should get removed soon.

Regarding `if/elsif/else` I do agree with you both, I am still used to go for one-liners where it is possible... I should get rid of that habit. :)

Reply:
> That would require to use an array of tags, which would prolong the execution time (more needles to evaluate). Or write more assert_screen(s) with a single tag.

Why would `assert_screen` here need an array? The calls are still the same, just the variable would be gone.

If you're referring to my other comment about keeping `jeos-system-locale-$lang`, I don't really see an issue there as we only have four `jeos-locale` needles and it would only be relevant for aarch64 in the future anyway.

Reply:
> > That would require to use an array of tags, which would prolong the execution time (more needles to evaluate). Or write more assert_screen(s) with a single tag.
> 
> Why would `assert_screen` here need an array? The calls are still the same, just the variable would be gone.
> 

Yup, my fault. I have edited my comment too late. Forgot to add it to the if/else... response. :)

> If you're referring to my other comment about keeping `jeos-system-locale-$lang`, I don't really see an issue there as we only have four `jeos-locale` needles and it would only be relevant for aarch64 in the future anyway.

Well, one of them had both tags. 
https://github.com/os-autoinst/os-autoinst-needles-opensuse/pull/686/files#diff-9473ffd361d39e662d308bdf6bc588a7
Although I do agree that it is not a troublemaker at the moment, I like the idea how the test handles keyboard layout setup. First match whether the test has reached the menu, and afterwards try to select the item from it. 
But maybe it would be enough just to sort out the needles. :)